app/assets/javascripts/index/directions-route-output.js

```diff
@@ -13,34 +13,49 @@ OSM.DirectionsRouteOutput = function (map) {
     weight: 12
   });
 
+  let distanceUnits = "km";
   let downloadURL = null;
 
-  function formatTotalDistance(m) {
-    if (m < 1000) {
-      return OSM.i18n.t("javascripts.directions.distance_m", { distance: Math.round(m) });
-    } else if (m < 10000) {
-      return OSM.i18n.t("javascripts.directions.distance_km", { distance: (m / 1000.0).toFixed(1) });
+  function translateDistanceUnits(m) {
+    if (distanceUnits === "km") {
+      return [m, "m", m / 1000, "km"];
     } else {
-      return OSM.i18n.t("javascripts.directions.distance_km", { distance: Math.round(m / 1000) });
+      return [m / 0.3048, "ft", m / 1609.344, "mi"];
     }
   }
 
-  function formatStepDistance(m) {
-    if (m < 5) {
+  function formatTotalDistance(minorValue, minorName, majorValue, majorName) {
+    const scope = "javascripts.directions.distance_in_units";
```

Comment:
```suggestion
  const formatters = {
    km: { units: m => [m, "m", m / 1000, "km"] },
    mi: { units: m => [m / 0.3048, "ft", m / 1609.344, "mi"] }
  };
  let currentFormatter = formatters.km;
  let downloadURL = null;

  function formatTotalDistance(m) {
    const [minorValue, minorName, majorValue, majorName] = currentFormatter.units(m);
    const scope = "javascripts.directions.distance_in_units";
```
so the `...translateDistanceUnits(...)` can be dropped

Reply:
If we ignore that what's called a *formatter* here doesn't format anything, I'm not going to do functions/objects as values of km/mi preference until I decide where I'm storing that preference. Let's suppose the preference is persisted in a cookie. Certainly I'm not going to store a function or an object there.

Reply:
Why would you want to store a function in a cookie?
`formatterUnits[Cookies.get("distanceUnits")]` or `formatterUnits[localStorage.distanceUnits]` works just fine.